### DISCUSSION TITLE
Why does the render.py script doesn't exactly follow the camera path

### DISCUSSION THREAD

**Discussion by sqartox** [NVlabs/instant-ngp#622] [Q&A] [answered]
Hi,
I'm currently trying to get a video from some scenes I have but whenever I use the render script it doesn't exactly follow the camera path I've set. When "previewing" the camera path while editing it everything seems okay but when I run the script some camera positions are missing.
Example: I'm trying to do 180° from left to right, then zoom in and reverse 180° to starting position (while zoomed). In the preview it works perfectly but when rendering the video I only have like 120/140° and then it reverse while zooming at the same time and doesn't fully come back to starting position.
I've tried settings multiple keyframes to insist on certain points but it doesn't change anything.
Am I missing something?

**Comment by francescofugazzi** [comment]
I've been able to obtain kinda perfect loops by setting 3 times the first camera (so three cameras in the same position) and then repeating that also for the last camera in the same position of the first one, again 3 times. Because of that many extra frames are rendered but then i can delete some of them. I hope it can be usefull

**Comment by Tom94** [comment]
Any chance you have camera smoothing enabled? That'd be applied on top of the loaded camera path, leading to the camera "dragging behind" (albeit moving more smoothly). You can ensure it's off via `testbed.camera_smoothing = False`.

**Reply by francescofugazzi** [reply]
Hi Tom, i can keep the smoothing to true by applying that workaround i described a while ago

**Reply by sqartox** [reply]
I'm using the following script: https://github.com/bycloudai/instant-ngp-Windows/blob/master/scripts/render.py (copied it to my folder)
I'm then running the following command:
python .\scripts\render.py --scene "path" --fps 300 --width 256 --height 144 --render_name "name"
I'm using such low resolution so the rendering is faster

**Reply by francescofugazzi** [reply]
I'm also using it

**Reply by sqartox** [reply]
Can't get it to work as I want even with multiple keyframes at the same place. How do you disable the smoothing while using this script?

**Reply by sqartox** [reply]
The problem I have is when I try to "go back" (not continuing the motion the same way) it doesn't do the full motion.

**Reply by francescofugazzi** [reply]
I quickly recompressed this video for showing that with method i obtain first and last postition identical for having a loop

https://user-images.githubusercontent.com/96065025/169660200-2f4902e9-fadb-48f4-a3a9-c4ba5883b418.mp4

**Reply by sqartox** [reply]
May I ask how do you make a video this long? Do you just modify the "fps" parameter or is there something else I am missing?

**Reply by meriemjabri** [reply]
> I quickly recompressed this video for showing that with method i obtain first and last postition identical for having a loop
> 
>  ricky_bench_dunes_test_2_540p.mp4
great job! amazing video. Did you manage to get rid of the blur, especially at the beginning of the video on the right and left sides or do you know what is the reason behind it ?

**Comment by francescofugazzi** [comment, marked as answer]
the fps keep it low like 60 or less, otherwise you will have tons of frames. I made a long path and then trying with the gui i kept track of the length messing with speed parameter and obtaining more or less what i wanted and then i used the parameter of how many seconds i wanted in the render script
________________________________
Da: sqartox ***@***.***>
Inviato: Saturday, May 21, 2022 6:15:56 PM
A: NVlabs/instant-ngp ***@***.***>
Cc: Francesco Fugazzi ***@***.***>; Comment ***@***.***>
Oggetto: Re: [NVlabs/instant-ngp] Why does the render.py script doesn't exactly follow the camera path (Discussion #622)


May I ask how do you make a video this long? Do you just modify the "fps" parameter or is there something else I am missing?

**Reply by sqartox** [reply]
Oh... I feel so stupid right now. I didn't see the "n_seconds" parameter. That's the thing I was missing this entire time.
Thank you for your time :)

**Comment by Carnarts** [comment]
I'm getting this problem too, it's not rendering all the frames for the camera path, will try duplicating the start and end keyframes. Also the most frustrating thing is manipulating the cameras, if I try and rotate them to face a direction with the rotation handles they spin around uncontrollably along with the rest of the scene. Is there a way to stop this? sometimes it seems to work but I'm not sure why it has to be so hyper sensitive.

**Reply by Carnarts** [reply]
Ok I get it now, you have to untwirl the world transform box to stop that behaviour, but it should have a better option for this ideally :)
